### PR TITLE
Revocation registry rotation scenarios

### PR DESCRIPTION
**Pull request by mirgee**
Signed-off-by: Miroslav Kovar <miroslavkovar@protonmail.com>

### CODE REVIEW

File: libvcx/src/aries/handlers/proof_presentation/prover/prover.rs [hunk 1]
```diff
@@ -34,7 +34,8 @@ impl Prover {
 
     pub fn retrieve_credentials(&self) -> VcxResult<String> {
         trace!("Prover::retrieve_credentials >>>");
-        let presentation_request = self.prover_sm.presentation_request().request_presentations_attach.content()?;
+        let presentation_request = self.prover_sm.presentation_request().request_presentations_attach.content()?; // TODO: How does this have known size at compile time?
```
Comment (Patrik-Stas) [inline libvcx/src/aries/handlers/proof_presentation/prover/prover.rs:37]: Why wouldn't it? Seems like it's just a string?

Comment (mirgee) [reply libvcx/src/aries/handlers/proof_presentation/prover/prover.rs:37]: This was meant to be 'How doesnt this have ...', I left it due to incorrect static analysis

File: libvcx/src/lib.rs [hunk 2]
```diff
@@ -659,4 +713,171 @@ mod tests {
...
+        let credential_data2 = json!({address1.clone(): "101 Tela Lane", address2.clone(): "Suite 1", city.clone(): "SLC", state.clone(): "WA", zip.clone(): "8721"}).to_string();
+        let credential_handle2 = _exchange_credential(credential_data2.clone(), cred_def_handle, consumer_to_issuer, issuer_to_consumer, Some(consumer), Some(req2));
+
+        revoke_credential(credential_handle1, rev_reg_id);
+
+        let proof_handle_verifier = _verifier_create_proof_and_send_request(&institution_did, &schema_id, &cred_def_id, verifier_to_consumer, Some(verifier), None);
+        _prover_select_credentials_and_send_proof(consumer_to_verifier, Some(consumer), None, Some(&credential_data2));
+        set_institution(Some(verifier));
+        proof::update_state(proof_handle_verifier, None, None).unwrap();
+        assert_eq!(proof::get_proof_state(proof_handle_verifier).unwrap(), ProofStateType::ProofValidated as u32);
```
Comment (Patrik-Stas) [inline libvcx/src/lib.rs:768]: perhaps we could check verification for both issued credentials? That would apply to other tests as well

File: libvcx/src/lib.rs [hunk 3]
```diff
@@ -659,4 +713,171 @@ mod tests {
...
+    }
+
+    #[test]
+    #[cfg(feature = "agency_pool_tests")]
+    fn test_two_creds_one_rev_reg_revoke_second() {
```
Comment (Patrik-Stas) [inline libvcx/src/lib.rs:773]: hmm seems like classic revocation scenario, don't we already have this somewhere else?

Comment (mirgee) [reply libvcx/src/lib.rs:773]: No, not with one holder.

**Review by Patrik-Stas — APPROVED** [review]

**Pull request merged**